common/iterator/ErrorHandledIterator.java

```diff
@@ -0,0 +1,43 @@
+package grakn.core.common.iterator;
+
+import grakn.core.common.exception.GraknException;
+
+import java.util.function.Function;
+
+public class ErrorHandledIterator<T> implements ResourceIterator<T> {
```

Comment:
Added `ErrorHandledIterator` so that when exception happens even when we're iterating through the list, we will still be able to close the transaction.